src/main/java/apoc/graph/document/builder/DocumentToGraph.java

```diff
@@ -28,10 +29,15 @@ public DocumentToGraph(GraphDatabaseService db, GraphsConfig config) {
     }
 
     public void validate(Map map) {
+        List<String> messages = new ArrayList<>();
         if (!map.containsKey(config.getIdField())) {
-            throw new RuntimeException("every object must have " + config.getIdField() + " as `id` field name");
-        } else if (!map.containsKey(config.getLabelField())) {
-            throw new RuntimeException("every object must have " + config.getLabelField() + " as `label` field name");
+            messages.add("`" + config.getIdField() + "` as id-field name");
+        }
+        if (!map.containsKey(config.getLabelField())) {
+            messages.add("`" + config.getLabelField() + "` as label-field name");
+        }
+        if (!messages.isEmpty()) {
+            throw new RuntimeException("The object " + JsonUtil.writeValueAsString(map) + " must have " + String.join(" and ", messages));
```

Comment:
we have to be careful, if the document is 100MB, either take a substring or render just the first few fields.
perhaps write to a string writer that's limited to a 1000 char buffer and then stop writing.